telemetry/telemetry.go

```diff
@@ -211,121 +229,207 @@ func isFlagPassed(fs *flag.FlagSet, name string) bool {
 	return found
 }
 
-func signalHandler(serverControlSignal chan<- int, wg *sync.WaitGroup, sigchannel <-chan os.Signal) {
-	defer wg.Done()
+func iNotifyCertMonitoring(watcher *fsnotify.Watcher, telemetryCfg *TelemetryConfig, serverControlSignal chan<- ServerControlValue, testReadySignal chan<- int) {
+	defer watcher.Close()
+
+	done := make(chan bool)
+
+	go func() {
+		if testReadySignal != nil { // for testing only
+			testReadySignal <- 0
+		}
+		for {
+			select {
+			case event := <-watcher.Events:
+				if event.Name != "" && (filepath.Ext(event.Name) == ".cert" || filepath.Ext(event.Name) == ".crt" ||
+                                   filepath.Ext(event.Name) == ".pem" || filepath.Ext(event.Name) == ".key") {
```

Comment:
This line is using space instead of tab?

Reply:
Fixed